### PR TITLE
BUG: Check for initial state finiteness

### LINKED ISSUE #17846: BUG: Infinite loop in scipy.integrate.solve_ivp()
### Describe your issue.

`scipy.integrate._ivp.common.select_initial_step()` returns `numpy.nan` if `max(abs(y0)) == numpy.inf` 

This causes programs to lock, when calling `solve_ivp()` with such an initial value.

Probably, `select_initial_step()` (or its callers) should raise an Error in that situation.

### Reproducing Code Example

```python
Call `solve_ivp()` with at least one component of `y0` being infinite (probably a NaN component would cause the same problem).
```

### Error message

```shell
None, program goes into infinite loop, when calling `solve_ivp()`
```

### SciPy/NumPy/Python version information

1.9.3 1.23.5 sys.version_info(major=3, minor=10, micro=6, releaselevel='final', serial=0)

### PR DESCRIPTION
**Pull request by Patol75**
#### Reference issue

Closes gh-17846

#### What does this implement/fix?

Providing an initial state `y0` containing non-finite array elements to `solve_ivp` causes execution to stall. Here, I add a simple check within `check_arguments` in `base.py` to interrupt the execution if `y0` contains non-finite values. 

#### Additional information

None.

### CODE REVIEW

**Comment by j-bowhay** [comment]
I think it would be desirable to add a test alongside this change

File: scipy/integrate/_ivp/base.py [hunk 1]
```diff
@@ -15,6 +15,9 @@ def check_arguments(fun, y0, support_complex):
 
     if y0.ndim != 1:
         raise ValueError("`y0` must be 1-dimensional.")
+    
+    if not np.isfinite(y0).all():
+        raise ValueError("All components of the initial state `y0` must be finite.")
```
Comment (tylerjereddy) [inline scipy/integrate/_ivp/base.py:20]: For the matching regression test, this seems to hang on `main` and pass instantly here:

```diff
diff --git a/scipy/integrate/_ivp/tests/test_ivp.py b/scipy/integrate/_ivp/tests/test_ivp.py
index 2fb294823..0ed0d752a 100644
--- a/scipy/integrate/_ivp/tests/test_ivp.py
+++ b/scipy/integrate/_ivp/tests/test_ivp.py
@@ -1038,3 +1038,9 @@ def test_args_single_value():
 
     sol = solve_ivp(fun_with_arg, (0, 0.1), [1], args=(-1,))
     assert_allclose(sol.y[0, -1], np.exp(-0.1))
+
+
+@pytest.mark.parametrize("f0_fill", [np.nan, np.inf])
+def test_gh_17846(f0_fill):
+    with pytest.raises(ValueError):
+        solve_ivp(fun_zero, [0, 10], np.full(3, f0_fill))
```

Comment (Patol75) [reply scipy/integrate/_ivp/base.py:20]: I am not sure I understand what you mean. I have compiled the branch associated with the PR, and calling `solve_ivp` with non-finite `y0` did raise a `ValueError`.

**Comment by tylerjereddy** [comment]
> I am not sure I understand what you mean. I have compiled the branch associated with the PR, and calling solve_ivp with non-finite y0 did raise a ValueError.

Sorry if I wasn't clear--I simply wrote a draft regression test in case it helps you address the comments from the core team. I think we're agreeing here.

File: scipy/integrate/_ivp/base.py (hunk 1 above)
Comment (j-bowhay) [inline scipy/integrate/_ivp/base.py:20]:
```suggestion

    if not np.isfinite(y0).all():
        raise ValueError("All components of the initial state `y0` must be finite.")
```

**Review by j-bowhay — APPROVED** [review]
Thanks, @Patol75 @tylerjereddy I think this is good to go. MATLAB will return just nans/infs if passed these as the initial conditions however I think being explicit and raising an error is the correct thing to do in the context of SciPy.

**Pull request merged**